src/test/java/com/hatchways/assessment/api/ApplicationTests.java

```diff
@@ -0,0 +1,162 @@
+			}
+		}
+		return true;
+	}
+
+	// Test for status code for api/posts route
+	@Test
+	public void testPostAPI() throws Exception {
+		String url = createURLWithPort("/api/posts?tags=tech", "8080");
+		try {
+			assertEquals(200,getStatus(url));
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+	}
+
+	// Test for status code for api/ping route
+	@Test
+	public void testPingAPI() throws Exception {
+		String url = createURLWithPort("/api/ping", "8080");
+		try {
+			assertEquals(200,getStatus(url));
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+	}
+
+	// Test for proper error handling
+	@Test
+	void testErrorResponses() throws Exception {
+		String url = createURLWithPort("/api/posts?tags=tech&sortBy=rubbish", "8080");
+		JsonObject json = getResponseJson(url);
+		assertEquals("{\"error\":\"sortBy parameter is invalid\"}", json.toString());
+		url = createURLWithPort("/api/posts?tags=tech&sortBy=id&direction=rubbish", "8080");
+		json = getResponseJson(url);
+		assertEquals("{\"error\":\"direction parameter is invalid\"}", json.toString());
+	}
+
+	// Test for responses being sorted
+	@Test
+	void testSortedResponses() throws Exception {
+		String url = createURLforSortTest("id", "asc", "8080", "/api/posts?tags=tech,history");
+		JsonObject json = getResponseJson(url);
+		assertEquals(isSorted(json,"id", "asc"), true);
+
+		url = createURLforSortTest("id", "desc", "8080", "/api/posts?tags=tech,history");
+		json = getResponseJson(url);
+		assertEquals(isSorted(json,"id", "desc"), true);
+
+		url = createURLforSortTest("reads", "asc", "8080", "/api/posts?tags=tech,history");
+		json = getResponseJson(url);
+		assertEquals(isSorted(json,"reads", "asc"), true);
+
+		url = createURLforSortTest("reads", "desc", "8080", "/api/posts?tags=tech,history");
+		json = getResponseJson(url);
+		assertEquals(isSorted(json,"reads", "desc"), true);
+
+		url = createURLforSortTest("likes", "asc", "8080", "/api/posts?tags=tech,history");
+		json = getResponseJson(url);
+		assertEquals(isSorted(json,"likes", "asc"), true);
+
+		url = createURLforSortTest("likes", "desc", "8080", "/api/posts?tags=tech,history");
+		json = getResponseJson(url);
+		assertEquals(isSorted(json,"likes", "desc"), true);
+
+		url = createURLforSortTest("popularity", "asc", "8080", "/api/posts?tags=tech,history");
+		json = getResponseJson(url);
+		assertEquals(isSorted(json,"popularity", "asc"), true);
+
+		url = createURLforSortTest("popularity", "desc", "8080", "/api/posts?tags=tech,history");
+		json = getResponseJson(url);
+		assertEquals(isSorted(json,"popularity", "desc"), true);
+	}
+
+	@Test
+	void testUniqueEntries() throws Exception {
+		String url = createURLWithPort("/api/posts?tags=tech,history,design", "8080");
+		JsonObject json = getResponseJson(url);
+		assertEquals(isUnique(json), true);
+	}
+	// helper function to create url for all tests
+	private String createURLWithPort(String uri, String port) {
```

Comment:
Helper functions should all be together. You can even extract them into a separate Helper.java and import in this file.